content/spin/distributing-apps.md

```diff
@@ -1,104 +1,133 @@
 
-## Distributing Spin Applications Based on the OCI Distribution Specification
+Many cloud services offer public registries.  Examples include GitHub Container Registry, Docker Hub, or Amazon Elastic Container Registry.  These support both public and private distribution.  You can also run your own registry using open source software.
 
-Starting with [v0.8.0](https://github.com/fermyon/spin/releases/tag/v0.8.0), Spin supports distributing applications using existing container registry services such as GitHub Container Registry, Docker Hub, Azure ACR, or AWS ECR. This feature is experimental and will continue to evolve in future versions of Spin.
+## Logging Into a Registry
 
-While this feature is still in the experimental phase, the CLI will reuse the container registry authentication used by the Docker CLI, so you first have to log in to the registry service using an existing tool (such as `docker login`, or using the instructions provided by the registry service). In a future version, this will be improved using a Spin login command for the target registry.
+Before you can publish to a registry, or run applications whose registry artifacts are private, you must log in to the registry.  This example shows logging into the GitHub Container Registry, `ghcr.io`:
 
-Pushing an application to a registry:
-
-<!-- @nocpy -->
+<!-- @selectiveCpy -->
 
 ```bash
-$ spin registry push ghcr.io/radu-matei/spin-hello-world:v1
-Pushed "https://ghcr.io/v2/radu-matei/spin-hello-world/manifests/sha256:06b19f4394c59fe943140c9b59f083aefd4b53c6b632758523a2800d819a1575"
+$ spin registry login ghcr.io
 ```
 
-Then run the application using the registry reference:
+If you don't provide any options to `spin registry login`, it prompts you for a username and password.
 
-<!-- @nocpy -->
+### Logging In Using a Token
+
+In a non-interactive environment such as GitHub Actions, you will typically log in using a token configured in the environment settings, rather than a password.  To do this, use the `--password-stdin` flag, and `echo` the token value to the login command's standard input.  This example shows logging into GHCR from a GitHub action:
+
+<!-- @noCpy -->
 
 ```bash
-$ spin registry run ghcr.io/radu-matei/spin-hello-world:v1
+$ echo "${{ secrets.GITHUB_TOKEN }}" | spin registry login ghcr.io --username ${{ github.actor }} --password-stdin
```

Comment:
I think some extra escaping for the `${{...}}` values may be necessary here.  I'm currently seeing this section render as follows:
<img width="951" alt="Screenshot 2023-03-14 at 3 06 29 PM" src="https://user-images.githubusercontent.com/9322017/225136398-42a18090-68b9-4d04-b220-89426d64578d.png">

Reply:
Yikes, completely missed that.  Thanks!